chat-ui/src/DevCenterChatbot.tsx

```diff
@@ -526,29 +566,59 @@ const Message = ({
           : messageData.content}
       </LGMessage>
       {displaySuggestedPrompts && (
-        <div className={styles.message_prompts}>
-          <MessagePrompts label="Suggested Prompts">
-            {suggestedPrompts.map((sp, idx) => (
-              <MessagePrompt
-                key={idx}
-                onClick={() => {
-                  setSuggestedPromptIdx(idx);
-                  setTimeout(() => {
-                    suggestedPromptOnClick(suggestedPrompts[idx]);
-                  }, 300);
-                }}
-                selected={idx === suggestedPromptIdx}
-              >
-                {sp}
-              </MessagePrompt>
-            ))}
-          </MessagePrompts>
-        </div>
+        <MessagePrompts
+          messagePrompts={suggestedPrompts}
+          messagePromptsOnClick={suggestedPromptOnClick}
+        />
       )}
     </Fragment>
   );
 };
 
+type MessagePromptsProps = {
+  messagePrompts: string[];
+  messagePromptsOnClick: (prompt: string) => void;
+};
+
+const MessagePrompts = ({
+  messagePrompts,
+  messagePromptsOnClick,
+}: MessagePromptsProps) => {
+  const [inProp, setInProp] = useState(true);
+  const [suggestedPromptIdx, setSuggestedPromptIdx] = useState(-1);
+  const nodeRef = useRef(null);
+  const duration = 300;
+
+  return (
+    <CSSTransition
+      in={inProp}
+      timeout={duration}
+      nodeRef={nodeRef}
+      classNames={styles.message_prompts(duration)}
+    >
+      <div className={styles.message_prompts(duration)} ref={nodeRef}>
+        <LGMessagePrompts label="Suggested Prompts">
+          {messagePrompts.map((sp, idx) => (
+            <MessagePrompt
+              key={idx}
```

Comment:
```suggestion
              key={sp}
```

In general you shouldn't use the map index as a key - note the second [Rule of Keys](https://react.dev/learn/rendering-lists#rules-of-keys)

> Keys must not change or that defeats their purpose! **Don’t generate them while rendering.**

We have a much better candidate key in the suggested prompt text itself - each one is unique and the text won't change between renders.